lib/hyrax/transactions/steps/delete_resource.rb

```diff
@@ -27,8 +27,8 @@ def initialize(persister: Hyrax.persister, publisher: Hyrax.publisher)
         def call(resource, user: nil)
           return Failure(:resource_not_persisted) unless resource.persisted?
 
-          @persister.delete(resource: resource)
           publish_changes(resource: resource, user: user)
+          @persister.delete(resource: resource)
```

Comment:
why? isn't it better to do the thing then publish that you did it?

Reply:
I agree, but fedora was complaining about tombstones. I'll double check this is still needed since there was another tombstone related fix in this PR.

Reply:
Still needed unless an alternative is found. Without the backwards order `find_parents` in `MemberCleanupListener` fails during `rspec spec/hyrax/transactions/work_destroy_spec.rb`:

```ruby
     Ldp::Gone:
       Discovered tombstone resource at /test/a1/38/5d/b3/a1385db3-8fbc-4330-b57d-7ffb972a5591, departed at: 2024-12-18T16:22:50.848357Z
     # /app/bundle/ruby/3.2.0/gems/ldp-1.2.1/lib/ldp/client/methods.rb:119:in `block in check_for_errors'
     # /app/bundle/ruby/3.2.0/gems/ldp-1.2.1/lib/ldp/client/methods.rb:117:in `check_for_errors'
     # /app/bundle/ruby/3.2.0/gems/ldp-1.2.1/lib/ldp/client/methods.rb:53:in `block in get'
     # /app/bundle/ruby/3.2.0/gems/activesupport-6.1.7.10/lib/active_support/notifications.rb:205:in `instrument'
     # /app/bundle/ruby/3.2.0/gems/ldp-1.2.1/lib/ldp.rb:43:in `instrument'
     # /app/bundle/ruby/3.2.0/gems/ldp-1.2.1/lib/ldp/client/methods.rb:33:in `get'
     # /app/bundle/ruby/3.2.0/gems/valkyrie-3.5.0/lib/valkyrie/persistence/fedora/query_service.rb:108:in `content_with_inbound'
     # /app/bundle/ruby/3.2.0/gems/valkyrie-3.5.0/lib/valkyrie/persistence/fedora/query_service.rb:42:in `find_parents'
     # ./app/services/hyrax/listeners/member_cleanup_listener.rb:19:in `on_object_deleted'
     # /app/bundle/ruby/3.2.0/gems/dry-events-1.0.1/lib/dry/events/bus.rb:46:in `call'
     # /app/bundle/ruby/3.2.0/gems/dry-events-1.0.1/lib/dry/events/bus.rb:46:in `block in publish'
     # /app/bundle/ruby/3.2.0/gems/dry-events-1.0.1/lib/dry/events/bus.rb:38:in `block in process'
     # /app/bundle/ruby/3.2.0/gems/dry-events-1.0.1/lib/dry/events/bus.rb:34:in `each'
     # /app/bundle/ruby/3.2.0/gems/dry-events-1.0.1/lib/dry/events/bus.rb:34:in `process'
     # /app/bundle/ruby/3.2.0/gems/dry-events-1.0.1/lib/dry/events/bus.rb:45:in `publish'
     # /app/bundle/ruby/3.2.0/gems/dry-events-1.0.1/lib/dry/events/publisher.rb:207:in `publish'
     # ./lib/hyrax/transactions/steps/delete_resource.rb:45:in `publish_changes'
```